# Lesson 2. 数值型对象的创建&emsp;<span style="color:blue">[DONE]</span>

&emsp;&emsp;数值类型对象主要包括：整数型（int）、浮点型（float）和复数型（complex）

## 1. 创建整数型对象

In [1]:
a = 2

In [2]:
# 使用 type 函数查看对象类型
type(a)

int

int 就是 Python 中的整数型对象。我们也可以用 int 函数来创建整数。

In [15]:
int(3)

3

In [16]:
int(3.1)

3

In [17]:
int(3.4)

3

int 函数的基本作用是将输入的数值取整后输出

## 2. 创建浮点型对象

简单理解，Python 浮点型对象就是数学意义中带小数位数的对象。

In [3]:
b = 3.14

In [4]:
b

3.14

In [5]:
type(b)

float

Python 中用 float 指代浮点型对象，我们也可以使用同名函数创建浮点型对象。

In [18]:
float(1)

1.0

In [1]:
# 使用科学计数法创建浮点型对象
c = 1.23e2

In [7]:
c

123.0

In [8]:
type(c)

float

在科学计数法中，e 代表以 10 为底，e 后面数字代表 10 的幂。  
例如：1.23e2 代表 1.23 * 100 = 123

**Point:**

- 通过科学计数法创建的数，即使是整数，也会用浮点型来表示；
- 浮点型也可用于存储整数数值。

In [9]:
f1 = 1

In [10]:
f2 = 1.0

## 3. 创建复数对象

Python 复数对象与数学意义中复数对象定义相同

In [2]:
# 创建一个实部为 3、虚部为 4 的复数
a =  3 + 4j

In [3]:
a

(3+4j)

In [4]:
type(a)

complex